exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -176,6 +181,23 @@ public void setAutoRead(boolean enableAutoRead) {
   }
 
   /**
+   * Sets the client name.
```

Comment:
Client name is more of the user agent, like JDBC client, ODBC client, C++ connector, etc...
For Java the application name is captured using the RuntimeMXBean, but maybe it should be overrideable too...